redis_json/src/ivalue_manager.rs

```diff
@@ -71,22 +71,21 @@ fn replace<F: FnMut(&mut IValue) -> Result<Option<IValue>, Error>>(
             }
             ValueType::Array => {
                 let arr = target_once.as_array_mut().unwrap();
-                if let Ok(x) = token.parse::<usize>() {
-                    if is_last {
-                        if x < arr.len() {
-                            let v = &mut arr.as_mut_slice()[x];
-                            if let Some(res) = (func)(v)? {
-                                *v = res;
-                            } else {
-                                arr.remove(x);
-                            }
+                let x = token.parse::<usize>().expect(
+                    "Array index should have been parsed successfully before reaching here",
```

Comment:
I think we can be a little more short:
```rust
"An array index is parsed successfully."
```
Or, we can return a `RedisError` from this function with a proper message, instead of panicking and so crashing down the server/node/shard.

Also, can we call it `index`, please?